controllers/kubeadm_control_plane_controller.go

```diff
@@ -195,3 +221,33 @@ func (r *KubeadmControlPlaneReconciler) reconcileDelete(_ context.Context, kcp *
 	controllerutil.RemoveFinalizer(kcp, clusterv1.KubeadmControlPlaneFinalizer)
 	return ctrl.Result{}
 }
+
+func (r *KubeadmControlPlaneReconciler) reconcileKubeconfig(ctx context.Context, clusterName, namespace string, endpoint clusterv1.APIEndpoint, kcp *clusterv1.KubeadmControlPlane) error {
+	if endpoint.IsZero() {
+		return nil
+	}
+
+	_, err := secret.GetFromNameAndNamespace(r.Client, clusterName, namespace, secret.Kubeconfig)
+	switch {
+	case apierrors.IsNotFound(err):
+		if createErr := kubeconfig.CreateSecretWithOwner(
```

Comment:
I find this a little easier to read without using the one line `if err := fn(); err != nil` syntax. My usual rule is if the function call is too long or wants to wrap, don't use the one line `if; err` syntax. But this is a nit, can ignore

Reply:
fixed.